machine_types.go

```diff
@@ -162,6 +162,15 @@ func (m *Machine) AllHealthChecks() *HealthCheckStatus {
 	return res
 }
 
+func (m *Machine) GetLatestEventOfType(eventType string) *MachineEvent {
+	for _, event := range m.Events {
+		if event.Type == eventType {
+			return event
+		}
+	}
+	return nil
+}
+
```

Comment:
I assume `m.Events` is always ordered by time descending?

Reply:
yea, i checked w Dov. We also make that assumption in `GetlatestEventOfTypeAfterType`